syntax = "proto3";

package api.v1;

option go_package = "api/v1"; // For language interoperability

message User {
  int64 id = 1;
  string username = 2;
  string email = 3;
}

message GetUserRequest {
  int64 user_id = 1;
}

message GetUserResponse {
  User user = 1;
}

message ListUsersRequest {
  int32 page = 1;
  int32 page_size = 2;
}

message ListUsersResponse {
  repeated User users = 1;
  int32 total = 2;
}

message InsertUserRequest {
  string username = 1;
  string email = 2;
}

// Define the Empty message here
message Empty {}

service UserService {
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
  rpc ListUsers(ListUsersRequest) returns (ListUsersResponse);
  rpc InsertUser(InsertUserRequest) returns (Empty);
}
